include/dlaf/auxiliary/norm/mc/norm_max_L.h

```diff
@@ -0,0 +1,107 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/hpx.hpp>
+#include <hpx/util/unwrap.hpp>
+
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/sync/reduce.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+namespace internal {
+namespace mc {
+
+// Compute max norm of the lower triangular part of the distributed matrix
+// https://en.wikipedia.org/wiki/Matrix_norm#Max_norm
+//
+// It just addresses tiles with elements belonging to the lower triangular part of the matrix
+//
+// Thanks to the nature of the max norm, it is valid for:
+// - ge/sy/he lower
+// - tr lower non-unit
+template <class T>
+dlaf::BaseType<T> norm_max_L(comm::CommunicatorGrid comm_grid, Matrix<const T, Device::CPU>& matrix) {
+  using dlaf::common::internal::vector;
+  using dlaf::common::make_data;
+  using hpx::util::unwrapping;
+
+  using dlaf::tile::lange;
+  using dlaf::tile::lantr;
+
+  using NormT = dlaf::BaseType<T>;
+
+  const auto& distribution = matrix.distribution();
+
+  DLAF_ASSERT_SIZE_SQUARE(matrix);
+  DLAF_ASSERT_BLOCKSIZE_SQUARE(matrix);
+
+  if (GlobalElementSize{0, 0} == matrix.size())
+    return {0};
+
+  vector<hpx::future<NormT>> tiles_max;
+  tiles_max.reserve(distribution.localNrTiles().rows() * distribution.localNrTiles().cols());
+
+  // for each tile in local (lower triangular), create a task that finds the max element in the tile
+  for (SizeType j_loc = 0; j_loc < distribution.localNrTiles().cols(); ++j_loc) {
+    const SizeType j = distribution.template globalTileFromLocalTile<Coord::Col>(j_loc);
+    const SizeType i_diag_loc = distribution.template nextLocalTileFromGlobalTile<Coord::Row>(j);
+
+    for (SizeType i_loc = i_diag_loc; i_loc < distribution.localNrTiles().rows(); ++i_loc) {
+      const SizeType i = distribution.template globalTileFromLocalTile<Coord::Row>(i_loc);
+
```

Comment:
Not really, it would be equivalent to add 
```
if (tile_wrt_global.row() < tile_wrt_global.col())
    continue;
```
but you will iterate on all the tiles of the matrix (adding also an if statement) instead of only half of them.

Reply:
> but you will iterate on all the tiles of the matrix (adding also an if statement) instead of only half of them.

Yes, that is the case. I was thinking more from a readability point of view as the cost of these is insignificant compared to the dataflow that follows.

> Not really, it would be equivalent to add

The comment about mentioning the `local` part threw me off a little bit. Perhaps then this would work then :

```
for (auto tile_wrt_local : iterate_range2d(dist.localNrTiles())) {
  auto tile_wrt_global = globalTileIndex(tile_wrt_local);
   // If the tile is not in the lower triangular part, skip it
   if (tile_wrt_global.row() < tile_wrt_global.col())
    continue;
  
   bool is_diag = tile_wrt_global.row() == tile_wrt_global.col();
  ...
}
```
or sth similar.

Reply:
Good point, the comment is incorrect... Yes this would work.

Reply:
I agree with you, I didn't use it because I didn't want to iterate over all tiles and skip. It was a premature optimization.

> EDIT: We can also define separate iterators for iterating over the lower/upper triangular parts of a region that avoid the costs of the if() statement and shorten the iteration range. Perhaps that is something to consider in the long run.

It is not urgent, but I think it would be a nice thing.

> The comment above mentioning the local part threw me off a little bit.

Yes, sorry it was misleading. I fixed it as @rasolca suggested 😉